homestar-runtime/src/network/swarm.rs

```diff
@@ -19,7 +18,11 @@ use std::{fmt, time::Duration};
 
 /// Build a new [Swarm] with a given transport and a tokio executor.
 pub async fn new(settings: &settings::Node) -> Result<Swarm<ComposedBehaviour>> {
-    let keypair = Keypair::generate_ed25519();
+    let keypair = settings
+        .network
+        .keypair_config
+        .generate_keypair()
+        .with_context(|| "Failed to generate/import keypair for libp2p".to_string())?;
```

Comment:
minor: have been not capitalizing error strings (as per how rust typically does it too).

Reply:
sounds good